app/src/main/java/at/roteskreuz/stopcorona/model/receivers/ExposureNotificationBroadcastReceiver.kt

```diff
@@ -16,24 +16,55 @@
  */
 package at.roteskreuz.stopcorona.model.receivers
 
+import android.app.NotificationManager
 import android.content.BroadcastReceiver
 import android.content.Context
 import android.content.Intent
+import androidx.core.app.NotificationCompat
+import androidx.work.WorkManager
+import at.roteskreuz.stopcorona.BuildConfig
+import at.roteskreuz.stopcorona.R
+import at.roteskreuz.stopcorona.model.workers.ProcessDiagnosisKeysWorker
 import com.google.android.gms.nearby.exposurenotification.ExposureNotificationClient
+import org.koin.standalone.KoinComponent
+import org.koin.standalone.inject
 import timber.log.Timber
 
 /**
  * Broadcast receiver for callbacks from exposure notification API.
  */
-class ExposureNotificationBroadcastReceiver : BroadcastReceiver() {
+class ExposureNotificationBroadcastReceiver : BroadcastReceiver(), KoinComponent {
+
+    private val workManager: WorkManager by inject()
+
     override fun onReceive(context: Context, intent: Intent) {
         val action = intent.action
         if (ExposureNotificationClient.ACTION_EXPOSURE_STATE_UPDATED == action) {
             val token = intent.getStringExtra(ExposureNotificationClient.EXTRA_TOKEN)
-            Timber.d("got a token '%s' which could be infected, we should check it", token)
+            Timber.d("got a token '%s' tp process the diagnosis keys, we should check it", token)
+
+            if (BuildConfig.DEBUG) {
+                showDebugNotificationProcessingFinished(context, token)
+            }
+
+            ProcessDiagnosisKeysWorker.enqueueProcessingOfDiagnosisKeys(workManager, token)
+
             //TODO process the exposure notification
             // see https://github.com/google/exposure-notifications-android/blob/master/app/src/main/java/com/google/android/apps/exposurenotification/nearby/StateUpdatedWorker.java
             // and https://github.com/google/exposure-notifications-android/blob/master/app/src/main/java/com/google/android/apps/exposurenotification/nearby/ExposureNotificationBroadcastReceiver.java
         }
     }
+
+    private fun showDebugNotificationProcessingFinished(context: Context, token: String) {
+        val notification = NotificationCompat.Builder(context, "channel_automatic_detection")
```

Comment:
Even for the debug functionality, I would suggest to move this code to the `NotificationRepository`. 
Btw. you didn't register the channel correctly, no notification will be shown.

Reply:
@falkorichter